lib/northbound_cli.c

```diff
@@ -276,14 +240,48 @@ int nb_cli_apply_changes(struct vty *vty, const char *xpath_base_fmt, ...)
 	 * faster.
 	 */
 	if (frr_get_cli_mode() == FRR_CLI_CLASSIC) {
-		if (vty->t_pending_commit || nb_cli_backoff_start(vty))
+		if (clear_pending) {
```

Comment:
Suppose we have some pending commands and this function is called with `clear_pending == false` and `vrf->pending_allowed == false`. In this case, we'll execute the provided command without executing the pending commands. Even if this combination of flags is not possible at the moment, I think the code should not allow this to happen anyway.

Reply:
If pending_allowed is false then there will never be any pending or it's violating that flag, this is an assertable condition I think and is fine. If someone is changing pending_allowed from true to false then they better d* flush the pending commits. :)

Reply:
Let's then add an `assert(!vty->pending_commit)` before `return nb_cli_classic_commit(vty);` to make sure that we quickly catch new bugs in the code.

Reply:
done.